core/src/main/java/com/amplifyframework/core/model/AuthRule.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core.model;
+
+import androidx.core.util.ObjectsCompat;
+
+import java.util.Arrays;
+
+/**
+ * {@link AuthRule} is used define an authorization rule for who can access and operate against a
+ * {@link Model} or a {@link ModelField}.
+ *
+ * @see <a href="https://docs.amplify.aws/cli/graphql-transformer/directives#auth")>GraphQL Transformer @auth directive
+ * documentation.</a>
+ */
+public final class AuthRule {
+    private final AuthStrategy allow;
+    private final String ownerField;
+    private final String identityClaim;
+    private final String groupClaim;
+    private final String[] groups;
+    private final String groupsField;
+    private final ModelOperation[] operations;
+
+    /**
+     * Constructor to create an {@link AuthRule} from an {@link com.amplifyframework.core.model.annotations.AuthRule}
+     * annotation.
+     * @param authRule an {@link com.amplifyframework.core.model.annotations.AuthRule} annotation.
+     */
+    public AuthRule(com.amplifyframework.core.model.annotations.AuthRule authRule) {
+        this.allow = authRule.allow();
+        this.ownerField = authRule.ownerField();
+        this.identityClaim = authRule.identityClaim();
+        this.groupClaim = authRule.groupClaim();
+        this.groups = authRule.groups();
+        this.groupsField = authRule.groupsField();
+        this.operations = authRule.operations();
+    }
+
+    /**
+     * Returns the type of strategy for this {@link AuthRule}.
+     * @return the type of strategy for this {@link AuthRule}
+     */
+    public AuthStrategy getAllow() {
+        return this.allow;
```

Comment:
Why not:
```
public AuthStrategy getAuthStrategy() {
    return this.authStrategy;
}
```

Reply:
For the @AuthRule annotation, `allow` is used since that is the term used in the schema.graphql.  For this model though, I agree `authStrategy` makes more sense, and have made this change.